include/eve/traits/overload/supports.hpp

```diff
@@ -0,0 +1,267 @@
+  using tag_of = std::remove_cvref_t<decltype(Func)>;
+
+  //====================================================================================================================
+  //! @addtogroup extensions
+  //! @{
+  //!   @struct decorated_with
+  //!   @brief Helper class to aggregate options handlers and states
+  //!
+  //!   **Defined in Header**
+  //!
+  //!   @code
+  //!   #include <eve/module/core.hpp>
+  //!   @endcode
+  //!
+  //!   EVE's @callable can be decorated with options. eve::decorated_with is used to list all subset of options a
+  //!   given EVE's @callable is allowed to support, how to handle the default values of such subsets and how to store
+  //!   said options states.
+  //!
+  //!   @tparam OptionsValues Options state type
+  //!   @tparam Options       List of options specifications to support
+  //!
+  //!   @groupheader{Example}
+  //!
+  //!   @subgroupheader{Interaction with eve::conditional}
+  //!   @godbolt{doc/traits/callable_supports.cpp}
+  //!
+  //!   @subgroupheader{Interaction with custom specifications}
+  //!   @godbolt{doc/traits/callable_specs.cpp}
+  //! @}
+  //====================================================================================================================
+  template< typename OptionsValues, typename... Options>
+  struct decorated_with : OptionsValues, Options...
+  {
+    using Options::process...;
+
+    constexpr EVE_FORCEINLINE decorated_with() {}
+    constexpr EVE_FORCEINLINE decorated_with(OptionsValues v) : OptionsValues(std::move(v)) {}
+
+    //==================================================================================================================
+    //! @brief Adds an option to current callable
+    //!
+    //! If the option `o` is supported by one of the specifications `Spec`, build a new @callable that will behave as
+    //! an instance of `Tag` except the option `o` is passed inside an eve::options as an additional parameter.
+    //!
+    //! Does not participate in overload resolution if `o` is not supported by any option specifications.
+    //!
+    //! @param o Option to add to current's @callable setup
+    //! @return A new @callable with the options `o` set.
+    //==================================================================================================================
+    template<typename O>
+    auto operator[](O o) const
+    requires( requires(OptionsValues const& ov) { this->process(ov,o);} )
+    {
+      return process(self(), o);
+    }
+
+    /// Retrieves the current options' state, including processed default
+    decltype(auto) options() const
+    {
+      return kumi::fold_left( [&](auto acc, auto const& m) { return m.default_to(acc); }
+                            , kumi::tuple<Options...>{}
+                            , self()
+                            );
+    }
+
+    // Short-cut to current state values
+    decltype(auto) self() const { return static_cast<OptionsValues const&>(*this); }
```

Comment:
Seems like not necessary and is a bit weird. Like - will my callable have `self()` method?

Reply:
Removed